mava/components/jax/building/data_server.py

```diff
@@ -136,6 +141,21 @@ def name() -> str:
         """Static method that returns component name."""
         return "data_server"
 
+    @staticmethod
+    def required_components() -> List[Type[Callback]]:
+        """List of other Components required in the system for this Component to function.
+
+        BaseSystemInit required to set up builder.store.agent_net_keys
+        and for config network_sampling_setup.
+        EnvironmentSpec required to set up builder.store.environment_spec
+        and builder.store.extras_spec.
+        AdderSignature required to set up builder.store.adder_signature_fn.
+
+        Returns:
+            List of required component classes.
+        """
+        return [EnvironmentSpec, AdderSignature, BaseSystemInit]
```

Comment:
with regards to the diagram showing 32 (ParallelSequenceAdderSignature) as having no dependencies, isn't that misleading? since the data_server will access the signature_fn from the store, which would either be set by the  transition or sequence adder signature component, depending on what component the system has.

Reply:
The diagram is a little confusing with both abstract classes and subclasses are shown. For example, since the MAPPO system has parallelSequenceAdder, it should have an implied dependency on ParallelSequenceAdderSignature

Reply:
Please add comments #todo to look in those redundant components. Perhaps issues are overkill

Reply:
We discussed `ParallelSequenceAdder` but we should revisit the necessity of the `ExecutorInit` and I think `ExtrasLogProbs`

Reply:
Created an issue @mmorris44 
https://github.com/instadeepai/Mava/issues/679